File: microservices/find/models/SupplierProduct.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;


const SupplierProductSchema = Schema({
    iso:{type: String, required: true},
    tenant_name: {type:String, required: true},
    tenant_id: {type:String, required: true, index: true},
    category_name: {type:String, required: true},
    category_display_name: {type:String, required: true},
    category_slug: {type:String, required: true},
    linked: {type:String, required: true, index: true}, // REFER
    supplier_category: {type:String, required: true, index: true}, // REFER
    shareable: {type:Boolean, default: true},
    published: {type:Boolean, default: true},
    object: {type:Object},
    created_at: {type:Date, default: Date.now, index: true},
});


module.exports = SupplierProduct = mongoose.model("supplier_products", SupplierProductSchema);

File: microservices/old-finder/models/supplierProduct.py
from database.db import db
from flask_mongoengine import BaseQuerySet
from models.category import Category
from models.supplierCategory import SupplierCategory
import datetime
# from slugify import slugify, Slugify, UniqueSlugify


class SupplierProduct(db.Document):
    iso = db.StringField(required=True)
    tenant_name = db.StringField(required=True)
    tenant_id = db.StringField(required=True)
    category_name = db.StringField(required=True)
    category_display_name = db.StringField(required=True)
    category_slug = db.StringField(required=True)
    linked = db.ReferenceField(Category)
    supplier_category = db.ReferenceField(SupplierCategory)
    shareable = db.BooleanField(default=True)
    published = db.BooleanField(default=True)
    object = db.ListField()
    created_at = db.DateTimeField(default=datetime.datetime.now, required=True)
    meta = {
        'indexes': ['-created_at','linked','supplier_category','tenant_id'],
        'ordering': ['-created_at'],
        'collection': 'supplier_products',
        'queryset_class': BaseQuerySet
    }

    def __setattr__(self, key, value):
        super(SupplierProduct, self).__setattr__(key, value)
        # if key == 'name':
        #     self.slug = slugify(self.name, to_lower=True)
    def __repr__(self):
        return '<SupplierProduct %r' % self.id